tuplex/codegen/src/IFailable.cc

```diff
@@ -32,4 +33,55 @@ void IFailable::logMessages() {
         else
             Logger::instance().defaultLogger().error(message);
     }
+}
+
+std::string IFailable::compileErrorToStr(const CompileError &err) {
+    std::string errMsg;
+    switch(err) {
```

Comment:
this is great. Hopefully we can add more errors & detailed err messages here in the future.